File: python/9.py
#!/usr/bin/env python

desired_sum = 1000
product = 0

for a in range(desired_sum):
    if product:
        break
    max_b = int((desired_sum - a) / 2)
    for b in range(a, max_b):
        c = 1000 - a - b
        if a**2 + b**2 == c**2:
            product = a * b * c
            break
print(product)

File: rust/src/bin/9.rs
fn main() {
    let desired_sum = 1000;
    let mut product = 0;
    let mut max_b;
    let mut c;

    for a in 0..desired_sum {
        if product != 0 {
            break;
        }
        max_b = (desired_sum - a) / 2;
        for b in a..max_b {
            c = 1000 - a - b;
            if a*a + b*b == c*c {
                product = a * b * c;
            }
        }
    }
    println!("{}", product);
}

